api/org/commcare/api/persistence/SqlHelper.java

```diff
@@ -79,13 +79,14 @@ public static PreparedStatement prepareTableSelectStatement(Connection c,
                                                                 String storageKey,
                                                                 String[] fields,
                                                                 String[] values,
-                                                                Persistable p) {
+                                                                Persistable p) throws IllegalArgumentException{
```

Comment:
`IllegalArgumentException` is a runtime exception, so it shouldn't be part of method signatures.